### DISCUSSION TITLE
Automatic prepare extensions in the project level `yaml` file

### DISCUSSION THREAD

**Discussion by mingsu** [quarto-dev/quarto-cli#3161] [Q&A] [answered]
To setup a project with several extensions or lua filter, I have to run the following command manually

```bash
quarto install extension quarto-ext/*
```

Will it be ok to specify the extension in the project `yaml` file, and automatically download the extensions when it is not exists?

**Comment by mcanouil** [comment, marked as answer]
Simply write a bash script in the pre-render option.
See <https://quarto.org/docs/projects/scripts.html#pre-and-post-render> for details.

```yaml
project:
  pre-render:
    - install_extensions.sh
```

Doing this, meaning by-passing the security questions is a security breach since you no longer ensure the code provided in the Lua filters/shortcodes is safe. (This code can access the file system)

**Reply by mingsu** [reply]
Thanks, understand now

**Reply by mcanouil** [reply]
If this answers your question, maybe update the category from "ideas" to "Q&A" and mark the the reply as the answer to let others know it has been answered.
